deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/AmqpsConnection.java

```diff
@@ -494,31 +463,45 @@ public boolean sendAmqpMessage(AmqpMessage message) throws Exception
     @Override
     public void onDelivery(Event event)
     {
-        logger.LogDebug("Entered in method %s", logger.getMethodName());
+        Link link = event.getLink();
 
-        AmqpMessage message = amqpDeviceOperations.receiverMessageFromLink(event.getLink().getName());
-        if (message == null)
+        if (link instanceof Sender)
         {
-            //Sender specific section for dispositions it receives
-            if (event.getType() == Event.Type.DELIVERY)
+            // Codes_SRS_AMQPSIOTHUBCONNECTION_15_038: [If this link is the Sender link and the event type is DELIVERY, the event handler shall get the Delivery (Proton) object from the event.]
+            Delivery d = event.getDelivery();
+            DeliveryState remoteState = d.getRemoteState();
+
+            // Codes_SRS_AMQPSIOTHUBCONNECTION_15_039: [The event handler shall note the remote delivery state and use it and the Delivery (Proton) hash code to inform the AmqpsIotHubConnection of the message receipt.]
+            boolean messageAcknowledgedAsSuccess = remoteState.equals(Accepted.getInstance());
```

Comment:
Previously, we ignored if our sent message was accepted or not. Now we fire a callback up to the listener to notify it if any message that was sent was rejected/abandoned/released/etc by the service so that it can stop waiting and propagate up the exception to the user